lib/streamlit/DeltaGenerator.py

```diff
@@ -1876,6 +1881,39 @@ def slider(
             current_value = current_value[0] if single_value else current_value
         return current_value if single_value else tuple(current_value)
 
+    @_with_element
+    def file_uploader(self, element, label, type=['*'], key=None):
+
+        """Display a file uploader widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this file uploader is for.
+        type : string[] or None
+            Array of allowed extensions. (['.png', '.jpg'])
+            Default value: ['*']
+
+        Returns
+        -------
+        byte[] or None
+            The byte array of uploaded file or None if no one file is loaded.
+
+        Examples
+        --------
+        >>> file = st.file_uploader("Upload a image", type=([".png"]))
+        >>> if file != None:
+        >>>     st.image(file)
+
+        """
+        element.file_uploader.label = label
+        element.file_uploader.type[:] = type
+        element.file_uploader.max_upload_size = config.get_option("server.maxUploadSize")
+        _set_widget_id("file_uploader", element, user_key=key)
+        progress, data = self._file_manager.get_data(element.file_uploader.id)
+        element.file_uploader.progress = progress
+        return data if data != None else NoValue
```

Comment:
Flip the order of this `if`, otherwise the `else` is a double-negative, which is hard to read.

Also, for `None` use `is not None` instead of `!= None`

```py
return NoValue if data is None else data
```